test/t/test_tmux.py

```diff
@@ -0,0 +1,167 @@
+        assert completion == ["bar", "bar bar.d/", "foo", "foo.d/"]
+
+    @pytest.mark.complete("tmux bind-key C-a ")
+    def test_subcommand_positional_arg_2(self, completion):
+        assert "new-session" in completion
+
+    @pytest.mark.complete("tmux start-server ")
+    def test_subcommand_no_positional_args(self, completion):
+        assert not completion
+
+    @pytest.mark.complete("tmux choose-tree abc def ghi ")
+    def test_subcommand_too_many_positional_args(self, completion):
+        assert not completion
+
+    # Tests for _comp_cmd_tmux()
+
+    @pytest.mark.complete("tmux -f ", cwd="shared/default")
+    def test_tmux_option_value(self, completion):
+        assert completion == ["bar", "bar bar.d/", "foo", "foo.d/"]
+
+    @pytest.mark.complete("tmux -")
+    def test_tmux_options(self, completion):
+        assert "-f" in completion  # takes a value
+        assert "-v" in completion  # takes no value
+
+    @pytest.mark.parametrize(
+        "other_commands",
+        [
+            r"foo \;",  # escaped in bash, not in tmux
+            r"foo\;",
+            r"foo \\\\\;",  # tmux: escaped backslash, unescaped semicolon
+            r"foo\\\\\;",
+            r"foo ';'",
+            r"foo '\\;'",  # tmux: escaped backslash, unescaped semicolon
+            r'foo ";"',
+            r'foo "\\\;"',  # tmux: escaped backslash, unescaped semicolon
+        ],
+    )
+    def test_tmux_multiple_commands(self, bash, other_commands):
+        completion = assert_complete(
+            bash,
+            f"tmux {other_commands} bind-key C-a ",
+        )
+        assert "new-session" in completion
+
+    @pytest.mark.parametrize(
+        "semicolon_arg",
+        [
+            r"foo \\\;",  # escaped in bash and tmux
+            r"foo\\\;",
+            r"foo '\;'",
+            r"foo'\;'",
+            r"foo '\\\;'",  # tmux: escaped backslash, escaped semicolon
+            r"foo'\\\;'",
+            r'foo "\;"',  # tmux: escaped semicolon
+            r'foo"\;"',
+            r'foo "\\;"',  # tmux: escaped semicolon
+            r'foo"\\;"',
+        ],
+    )
+    def test_tmux_semicolon_within_subcommand(self, bash, semicolon_arg):
+        completion = assert_complete(
+            bash,
+            # Note that the next arg is a file, not a subcommand. If it were a
+            # subcommand, it wouldn't be possible to tell if the completion was
+            # parsing semicolon_arg correctly.
+            f"tmux source-file {semicolon_arg} ",
+            cwd="shared/default",
+        )
+        assert completion == ["bar", "bar bar.d/", "foo", "foo.d/"]
+
+    @pytest.mark.xfail(reason="_comp_cmd_tmux() not called at all")
+    @pytest.mark.complete(r"tmux foo \; new-")
+    def test_tmux_subsequent_command_name_with_prefix(self, completion):
+        # TODO: figure out why _comp_cmd_tmux() isn't being called in this case
+        assert "new-session" in completion
```

Comment:
I think this is a bug in bash? https://savannah.gnu.org/support/?111224

Should I just remove this test case since the relevant code here is already covered by other test cases?

Reply:
The bug in bash was acknowledged, so I removed this test.